jax/_src/numpy/ufuncs.py

```diff
@@ -700,10 +700,12 @@ def sinc(x):
 def _sinc_maclaurin(k, x):
   # compute the kth derivative of x -> sin(x)/x evaluated at zero (since we
   # compute the monomial term in the jvp rule)
+  # TODO(mattjj): use a full_like primitive here rather than multiplying by
+  # zero, see https://github.com/google/jax/issues/10750
   if k % 2:
-    return lax.full_like(x, 0)
+    return x * 0
   else:
-    return lax.full_like(x, (-1) ** (k // 2) / (k + 1))
+    return x * 0 + lax.full_like(x, (-1) ** (k // 2) / (k + 1))
```

Comment:
We could remove the `full_like` in this line, and just rely on broadcasting, yes?

Reply:
I think it might insert a `convert_element_type` for us in some cases but I'm not sure. That is, I didn't want `(-1) ** (k // 2) / (k + 1)` to cause dtype promotion, and the `full_like` will ensure it's cast down to be like `x`. But I guess it's always weak?

Reply:
Maybe use `_lax_const(x, (-1) ** (k // 2) / (k + 1))` then?